scripts/linters/pylint_extensions.py

```diff
@@ -1960,6 +2079,388 @@ def process_tokens(self, tokens):
+                    if (
+                        import_status_dict['single_line_import'] and
+                        import_status_dict['import_line_num'] == line_num
+                    ):
+                        return
+                    elif import_status_dict['inside_multi_line_import_scope']:
+                        return
+                # Checking if comment for exceptional_type is present and it's
+                # with in the range (minimum 15 line of gaps).
+                if (
+                    type_status_dict['type_comment_no_longer_pending'] and
+                    line_num <= (
+                        type_status_dict['type_comment_line_num'] +
+                        ALLOWED_LINES_OF_GAP_IN_COMMENT
+                    )
+                ):
+                    type_status_dict['type_comment_no_longer_pending'] = False
+                else:
+                    self._add_exceptional_type_error_message(
+                        exceptional_type, line_num, node
+                    )
+
+    def _add_exceptional_type_error_message(
+        self, exceptional_type, line_num, node
+    ):
+        """This method should be called only when an exceptional type error is
+        encountered. If the exceptional type is Any then 'any-type-used' error
+        message is added, for object 'object-class-used' is added and for cast
+        'cast-func-used' is added.
+
+        Args:
+            exceptional_type: str. The exceptional type for which this method
+                is called, Possible values can be 'Any', 'object' and 'cast'.
+            line_num: int. The line number where error is encountered.
+            node: astroid.scoped_nodes.Module. Node to access module content.
+        """
+        if exceptional_type == 'Any':
+            self.add_message(
+                'any-type-used', line=line_num, node=node
+            )
+        if exceptional_type == 'object':
+            self.add_message(
+                'object-class-used', line=line_num, node=node
+            )
+        if exceptional_type == 'cast':
+            self.add_message(
+                'cast-func-used', line=line_num, node=node
+            )
+
+    def check_comment_is_present_with_object_class(self, tokens, node):
+        """Checks whether the object class in a module has been documented
+        or not. If the object class is not documented then adds an error
+        message.
+
+        Args:
+            tokens: Token. Object to access all tokens of a module.
+            node: astroid.scoped_nodes.Module. Node to access module content.
+        """
+        object_class_status_dict = copy.deepcopy(
+            self.EXCEPTIONAL_TYPE_STATUS_DICT
+        )
+
+        object_type_regex = r'Here we use object because'
+
+        for (token_type, token, (line_num, _), _, line) in tokens:
+            line = line.strip()
+
+            if token_type == tokenize.COMMENT:
+                if object_type_regex in line:
```

Comment:
Do you want to require that this be at the start of the line? If so, change it everywhere

Reply:
Ahh, sorry but can you please explain this comment a bit more, please!

Reply:
The current code will recognize the following as a valid documentation comment: `The following code does ... Here we use MyPy ignore because ...`. Do we really want to accept such a comment, or do we want to require that the `Here we use MyPy ignore because ...` part come at the start of the line so it's easier to identify. I'm not sure what the intended behavior is here

Reply:
We really want to require that the `Here we use MyPy ignore because ...` part come first, but there might be some cases where we are using TODOs. So, to allow todos as well I'm not forcing  `Here we use MyPy ignore because ...`  to be the start of the comment.